protocols/kad/src/behaviour.rs

```diff
@@ -217,6 +209,30 @@ pub enum Caching {
 }
 
 impl Config {
+    /// Builds a new `Config` with the given protocol names.
+    pub fn new(protocol_names: Vec<StreamProtocol>) -> Self {
```

Comment:
I'd prefer only a single protocol here going forward. If a user wants to support multiple DHTs, I think they should make multiple instances of `libp2p_kad::Behaviour`. I don't really see why we would need to merge those two into one record store.

Reply:
> I don't really see why we would need to merge those two into one record store.

My only guess would be to maintain compatibility with other networks without having to split the record store (although this probably wouldnt matter once #4817 or some better interface is sorted out) or having two or more kad behaviours running. Its fine for me either way (since i think go-libp2p and js-libp2p supports a single protocol for kad, but dont quote me on that)

Reply:
This seems to be on purpose
 https://github.com/libp2p/rust-libp2p/blob/fe29beee77d09f727b3e901798e9162dece50532/protocols/kad/src/protocol.rs#L152-L156

`ProtocolConfig` currently stores a `Vec<StreamProtocol>`
https://github.com/libp2p/rust-libp2p/blob/7730801d3ea2117c35dd222814928149dae9fc1b/protocols/kad/src/protocol.rs#L139-L144

## Alternative 1
If we want to accept a single protocol, we could either modify `new` so that it accepts a single `StreamProtocol` and keep the `Vec<StreamProtocol>` in `ProtocolConfig` (even though there will always be a single protocol).
https://github.com/libp2p/rust-libp2p/blob/7730801d3ea2117c35dd222814928149dae9fc1b/protocols/kad/src/protocol.rs#L147-L153

## Alternative 2
Or we can also modify the `ProtocolConfig` to store just 1 `StreamProtocol` (and not a vec). But this will cause other interfaces to break.
https://github.com/libp2p/rust-libp2p/blob/7730801d3ea2117c35dd222814928149dae9fc1b/protocols/kad/src/protocol.rs#L162-L165
https://github.com/libp2p/rust-libp2p/blob/7730801d3ea2117c35dd222814928149dae9fc1b/protocols/kad/src/protocol.rs#L167-L172
https://github.com/libp2p/rust-libp2p/blob/7730801d3ea2117c35dd222814928149dae9fc1b/protocols/kad/src/protocol.rs#L196-L198

____

I am in favor of either keep things as they are (`new(protocol_names: Vec<StreamProtocol>)`) or Alternative 1. IMO Alternative 2 breaks too much interfaces.

Reply:
Yes, definitely do this as a non-breaking change, i.e. Alternative 1.

My intention was that we can gradually deprecate this "set multiple protocols" concept so with the introduction of new APIs, we should only allow a single protocol going forward.

Reply:
Done! I think we are good to merge, unless you see something else?